src/App.tsx

```diff
@@ -3,58 +1,109 @@
 
-              <Loader />
+  const errorMessage = usersErrorMessage || postsErrorMessage;
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const handleSelectUserId = (user: User) => {
+    setIsLoading(true);
+    setSelectedPost(null);
+    setSelectedUser(user);
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+    if (selectedUser === user) {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  onSelectUser={handleSelectUserId}
+                />
               </div>
 
-              <PostsList />
+              <div className="block" data-cy="MainContent">
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
+
+                {selectedUser && isLoading && <Loader />}
+
+                {selectedUser && !isLoading && errorMessage && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    {errorMessage}
+                  </div>
+                )}
+
+                {selectedUser &&
+                  !isLoading &&
+                  !errorMessage &&
+                  posts.length === 0 && (
+                    <div
+                      className="notification is-warning"
+                      data-cy="NoPostsYet"
+                    >
+                      No posts yet
+                    </div>
+                  )}
+
+                {selectedUser &&
+                  !isLoading &&
+                  !errorMessage &&
+                  posts.length !== 0 && (
+                    <PostsList
+                      posts={posts}
+                      selectedPost={selectedPost}
+                      onSelectPost={setSelectedPost}
+                    />
+                  )}
+              </div>
             </div>
           </div>
-        </div>
 
-        <div
-          data-cy="Sidebar"
-          className={classNames(
-            'tile',
-            'is-parent',
-            'is-8-desktop',
-            'Sidebar',
-            'Sidebar--open',
-          )}
-        >
-          <div className="tile is-child box is-success ">
-            <PostDetails />
+          <div
+            data-cy="Sidebar"
+            className={cn('tile is-parent is-8-desktop Sidebar', {
+              'Sidebar--open': selectedPost,
+            })}
```

Comment:
The `className` for the sidebar uses `classnames` to conditionally apply `Sidebar--open`. Ensure that `selectedPost` is the correct condition for opening the sidebar. If there are other conditions or states that should affect this, consider adding them.